internal/controller/rolebindingtemplate_controller.go

```diff
@@ -0,0 +1,322 @@
+			expectedRB := &rbacv1.RoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      rolebinding.Name,
+					Namespace: ns,
+				},
+			}
+			_, err := ctrl.CreateOrUpdate(ctx, r.Client, expectedRB, func() error {
+				expectedRB.ObjectMeta = rolebinding.ObjectMeta
+				expectedRB.Subjects = rolebinding.Subjects
+				expectedRB.RoleRef = rolebinding.RoleRef
+				return ctrl.SetControllerReference(template, expectedRB, r.Scheme)
+			})
+			if err != nil {
+				log.Error(err, errCreateOrUpdateRB, "namespace", ns, "rolebinding", rolebinding.Name)
+				failedNamespaces[ns] = errCreateOrUpdateRB
+			} else {
+				provisionedNamespaces = append(provisionedNamespaces, ns)
+			}
+		}
+	}
+
+	template.Status.ProvisionedNamespaces = provisionedNamespaces
+	for ns, reason := range failedNamespaces {
+		template.Status.FailedNamespaces = append(template.Status.FailedNamespaces, v1alpha1.RoleBindingTemplateNamespaceFailure{
+			Namespace: ns,
+			Reason:    reason,
+		})
+	}
+
+	// > 0 failed namespaces means reconciliation was not successful
+	if len(template.Status.FailedNamespaces) > 0 {
+		err := errors.New(errNamespacesFailed)
+		log.Error(err, "failedNamespaces", template.Status.FailedNamespaces)
+		meta.SetStatusCondition(&template.Status.Conditions, metav1.Condition{
+			Type:    v1alpha1.ReadyCondition,
+			Status:  metav1.ConditionFalse,
+			Reason:  v1alpha1.FailedReason,
+			Message: errNamespacesFailed,
+		})
+		return ctrl.Result{}, err
+	}
+
+	log.Info(msgReconcileSucceeded, "name", req.Name, "provisionedNamespaces", provisionedNamespaces, "failedNamespaces", failedNamespaces)
+	meta.SetStatusCondition(&template.Status.Conditions, metav1.Condition{
+		Type:    v1alpha1.ReadyCondition,
+		Status:  metav1.ConditionTrue,
+		Reason:  v1alpha1.SucceededReason,
+		Message: msgReconcileSucceeded,
+	})
+	return ctrl.Result{}, nil
+}
+
+func (r *RoleBindingTemplateReconciler) getNamespacesFromScope(ctx context.Context, scopes v1alpha1.RoleBindingTemplateScopes) ([]string, error) {
+	labelSelector, err := metav1.LabelSelectorAsSelector(&scopes.OrganizationSelector)
+	if err != nil {
+		return nil, err
+	}
+
+	var organizations security.OrganizationList
+	if err := r.List(ctx, &organizations, &client.ListOptions{LabelSelector: labelSelector}); err != nil {
+		return nil, err
+	}
+
+	namespaces, err := r.getNamespacesFromOrganizations(ctx, &organizations)
+	if err != nil {
+		return nil, err
+	}
+
+	return namespaces, nil
+}
+
+func (r *RoleBindingTemplateReconciler) getNamespacesFromOrganizations(ctx context.Context, organizations *security.OrganizationList) ([]string, error) {
+	namespaces := []string{}
+	for _, o := range organizations.Items {
+		if o.Status.Namespace != "" {
+			// get the org namespace
+			namespaces = append(namespaces, o.Status.Namespace)
+		}
+
+		// get the cluster namespaces that belong to the org namespace
+		labelSelector, err := labels.Parse(fmt.Sprintf("%s=%s,%s", label.Organization, o.Name, label.Cluster))
```

Comment:
This feels like it should be defined elsewhere, but I don't know if it already is